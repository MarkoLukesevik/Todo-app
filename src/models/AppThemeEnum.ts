export enum AppThemeEnum {
    Light = 0,
    Dark = 1
}
