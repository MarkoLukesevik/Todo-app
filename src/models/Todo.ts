export interface Todo {
  id: string
  title: string
  isComplete: false
}
