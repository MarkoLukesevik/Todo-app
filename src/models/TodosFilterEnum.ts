export enum TodosFilterEnum {
    All = 0,
    Active = 1,
    Completed = 2
}