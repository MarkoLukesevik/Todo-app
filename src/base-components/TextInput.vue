<script setup lang="ts">
import { AppThemeEnum } from '@/models/AppThemeEnum';
import { computed } from 'vue';

const props = defineProps<{
    placeholder: string;
    modelValue: string;
    appTheme: AppThemeEnum;
}>();

const emit = defineEmits<{
    (e: 'update:modelValue', value: string): void;
}>();

const textInputColor = computed<string>(() => {
    return props.appTheme === AppThemeEnum.Light ? '#9495a5' : '#767992';
});

const handleInputChange = (event: Event): void => {
    emit('update:modelValue', (event.target as HTMLInputElement).value);
};
</script>

<template>
  <input
    class="app-input"
    :style="{ color: textInputColor }"
    :placeholder="placeholder"
    :value="modelValue"
    @input="handleInputChange($event)"
  />
</template>

<style scoped lang="scss">
.app-input {
    display: flex;
    flex-direction: row;
    flex: 1 1 auto;

    background-color: transparent;
    border: none;

    font-family: 'Josefin Sans', sans-serif;
    font-size: 0.9rem;
    font-weight: 400;

    &:focus {
        outline: none;
    }
}
</style>
