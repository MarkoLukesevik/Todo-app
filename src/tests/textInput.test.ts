import { describe, it, expect, beforeEach } from 'vitest';
import { shallowMount } from '@vue/test-utils';

import { AppThemeEnum } from '@/models/AppThemeEnum';

import TextInput from '@/base-components/TextInput.vue';

describe('AppInput', () => {
    let wrapper: ReturnType<typeof shallowMount>;

    beforeEach(() => {
        wrapper = shallowMount(TextInput, {
            props: {
                placeholder: 'Enter text',
                modelValue: 'initial value',
                appTheme: AppThemeEnum.Light,
            },
        });
    });

    it('Should match the snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });

    it('Should render correctly with default props', () => {
        const input = wrapper.find('input');
        expect(input.exists()).toBe(true);
        expect(input.attributes('placeholder')).toBe('Enter text');
        expect(input.element.value).toBe('initial value');
        expect(input.element.style.color).toBe('rgb(148, 149, 165)');
    });

    it('Should apply the correct text color based on appTheme prop', async () => {
        expect(wrapper.find('input').element.style.color).toBe('rgb(148, 149, 165)');
        
        await wrapper.setProps({ appTheme: AppThemeEnum.Dark });
        expect(wrapper.find('input').element.style.color).toBe('rgb(200, 203, 231)');
    });

    it('Should emit update:modelValue event with the input value when input changes', async () => {
        const input = wrapper.find('input');
        
        await input.setValue('new value');
        expect(wrapper.emitted('update:modelValue')).toBeTruthy();
        expect(wrapper.emitted('update:modelValue')?.[0]).toEqual(['new value']);
    });
});