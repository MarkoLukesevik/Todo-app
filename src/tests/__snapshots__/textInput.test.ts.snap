// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppInput > Should match the snapshot 1`] = `
<input
  class="app-input"
  data-v-4befb250=""
  placeholder="Enter text"
  style="color: rgb(148, 149, 165);"
  value="initial value"
/>
`;
